--- a/init-install-efi.sh	2020-01-02 15:08:43.000000000 +0100
+++ b/init-install-efi.sh	2020-01-23 11:57:34.071561022 +0100
@@ -15,9 +15,9 @@
     boot_size=$(( boot_size - $( du -ms /run/media/$1/$2 | awk '{print $1}') ))
 fi
 # remove initrd from size since its not currently installed
-if [ -e /run/media/$1/initrd ]; then
-    boot_size=$(( boot_size - $( du -ms /run/media/$1/initrd | awk '{print $1}') ))
-fi
+#if [ -e /run/media/$1/initrd ]; then
+#    boot_size=$(( boot_size - $( du -ms /run/media/$1/initrd | awk '{print $1}') ))
+#fi
 # add 10M to provide some extra space for users and account
 # for rounding in the above subtractions
 boot_size=$(( boot_size + 10 ))
@@ -241,8 +241,13 @@
 
 EFIDIR="/boot/EFI/BOOT"
 mkdir -p $EFIDIR
-# Copy the efi loader
+# Copy the efi loader, configuration files and certificates
 cp /run/media/$1/EFI/BOOT/*.efi $EFIDIR
+cp /run/media/$1/EFI/BOOT/*.inc $EFIDIR
+cp /run/media/$1/EFI/BOOT/*.p7b $EFIDIR
+if [ -f /run/media/$1/EFI/BOOT/grubenv ]; then
+    cp /run/media/$1/EFI/BOOT/grubenv $EFIDIR
+fi
 
 if [ -f /run/media/$1/EFI/BOOT/grub.cfg ]; then
     root_part_uuid=$(blkid -o value -s PARTUUID ${rootfs})
@@ -287,6 +292,13 @@
     done
 done
 
+# Copy initramfs artifacts. To add more artifacts just add to types
+for types in microcode initrd initramfs; do
+    for initrd in `find /run/media/$1/ -name $types*`; do
+        cp $initrd /boot
+    done
+done
+
 umount /boot
 
 sync
